File: LinkedList.java
public class LinkedList {
    private Node head;
    private Node tail;
    private int length;

    public LinkedList(double value) {
        this.head = new Node(value);
        this.tail = this.head;
        this.length = 1;
    }

    private static class Node {
        public double value;
        public Node next;
        Node(double value) {
            this.value = value;
            this.next = null;
        }
    }

    public void append(double value) {
        Node newNode = new Node(value);
        this.tail.next = newNode;
        this.tail = newNode;
        this.length++;
    }

    public void prepend(double value) {
        Node newNode = new Node(value);
        newNode.next = this.head;
        this.head = newNode;
        this.length++;
    }

    public void printList() {
        Node currentNode = this.head;
        System.out.print("HEAD -> ");
        while (currentNode != null) {
            System.out.print(currentNode.value + " -> ");
            currentNode = currentNode.next;
        }
        System.out.println("NULL");
    }

    public static void main(String[] args) {
        LinkedList myLinkedList = new LinkedList(10F);
        myLinkedList.append(5);
        myLinkedList.append(12.3);
        myLinkedList.append(18.3);
        myLinkedList.append(16);
        myLinkedList.prepend(1);
        myLinkedList.prepend(2);
        myLinkedList.prepend(3);

        myLinkedList.printList();
    }
}

File: LinkedList.js
class Node {
  constructor(value) {
    this.value = value;
    this.next = null;
  }
}

class LinkedList {
  constructor(value) {
    this.head = {
      value: value,
      next: null,
    };
    this.tail = this.head;
    this.length = 1;
  }

  append(value) {
    const newNode = new Node(value);

    this.tail.next = newNode;
    this.tail = newNode;
    this.length++;
  }

  prepend(value) {
    const newNode = new Node(value);

    newNode.next = this.head;
    this.head = newNode;
    this.length++;
  }

  insert(value, index) {
    let currentNode = this.head;
    let i = 0;
    while (currentNode !== null) {
      if (i === index) {
        currentNode.value = value;
        break;
      }
      i++;
      currentNode = currentNode.next;
    }
  }

  printList() {
    let currentNode = this.head;
    let str = "HEAD -> ";
    while (currentNode !== null) {
      str += `${currentNode.value} -> `;
      currentNode = currentNode.next;
    }
    str += "NULL";
    console.log(str);
  }
}

let myLinkedList = new LinkedList(10);

myLinkedList.append(5);
myLinkedList.append(16);
myLinkedList.prepend(1);
myLinkedList.insert(15, 2);
myLinkedList.printList();
